plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/types/ClassTypeEmbedding.kt

```diff
@@ -55,4 +55,18 @@ private fun PretypeEmbedding.isCollectionTypeNamed(name: String): Boolean {
         }
         return false
     }
-}
\ No newline at end of file
+}
+
+val ClassTypeEmbedding.isString: Boolean
+    get() = NameMatcher.matchGlobalScope(name) {
+        ifPackageName("kotlin") {
+            ifClassName("String") {
+                return true
+            }
+        }
+        return false
+    }
+
+fun ClassTypeEmbedding.embedClassTypeFunc(): DomainFunc =
+    if (isString) RuntimeTypeDomain.stringType
+    else RuntimeTypeDomain.classTypeFunc(name)
```

Comment:
Does the string class need this amount of special treatment?  I feel like it's a class like any other when it comes to RTTI

Reply:
I guess we could do the same with `String` not being special here.
We could do the same with `Boolean`s, `Int`s and `Char`s as well.
It may have some sense to reduce the amount of code we need to add a new primitive type.
On the other hand, it feels nice when these type embeddings have short beautiful names.
My thought regarding `String` was that it is treated somewhat special in Kotlin (having `String` literals etc.) and that is the main reason I decided to do it like this